### PR TITLE
[WIP] PSE management revamping

### LINKED ISSUE #1122: No way to display PSE image and documents in front-office
We have currently no way to display a Product Sale Element specific image and document in the front office.

A loop (or a parameter to image and document loop) is missing.

### PR DESCRIPTION
**Pull request by roadster31**
The main goal of this PR is to offer a clean and responsive alternative to the bloated, table-based PSE edition system currently in use.

A visual feedback for attached images, documents and for the PSE's virtual document has been added.

The document and image loops have been improved (`pse` `parameter), to provide a way to access PSE attached image and documents, and fix #1122. 

The document loop also provide a way to get the virtual file attached to a PSE, througth the `virtual_product` argument.

The PR also fixes a problem which prevent Thelia core translations to be saved by the "Translation" feature.

Some screenshots of the new 'Price' tab :

img-2015-04-12 15 32 49
img-2015-04-12 15 32 36
img-2015-04-12 15 32 13

### CODE REVIEW

**Comment by lovenunu** [comment]
Wow ! This seems very good !

**Comment by lunika** [comment]
The new foreign keys are missing in the update files. Complete the file 2.2.0-alpha1.sql.tpl and then generate the sql file.

Thanks

**Comment by roadster31** [comment]
Done !

File: core/lib/Thelia/Config/Resources/routing/admin.xml [hunk 1]
```diff
@@ -417,6 +417,10 @@
         <default key="_controller">Thelia\Controller\Admin\ProductController::deleteProductSaleElementAction</default>
     </route>
 
+    <route id="admin.product.combination.delete-all" path="/admin/product/combination/delete-all">
```
Comment (lunika) [inline core/lib/Thelia/Config/Resources/routing/admin.xml]: This route is vulnerable to CSRF attack, protect it with a token. Use all the form process or the url_token function in your template and verify it in your controller

**Pull request closed without merging**